test-framework/security/src/main/java/io/quarkus/test/security/QuarkusSecurityTestExtension.java

```diff
@@ -37,7 +36,10 @@ public void beforeEach(QuarkusTestMethodContext context) {
                     }).toArray(Class<?>[]::new));
             TestSecurity testSecurity = method.getAnnotation(TestSecurity.class);
             if (testSecurity == null) {
-                testSecurity = test.getAnnotation(TestSecurity.class);
+                testSecurity = original.getAnnotation(TestSecurity.class);
+                while (testSecurity == null && original != Object.class) {
+                    original = original.getSuperclass();
+                }
```

Comment:
Is this while loop missing an assignment to `testSecurity`? In other words, should that `testSecurity = original.getAnnotation(TestSecurity.class)` be moved into the while loop?

Reply:
Ah, oops. It still fixes it in this specific case as it is checking the original, however this was supposed to make it work for superclasses as well.